pkg/deploy/deploy.go

```diff
@@ -130,6 +128,14 @@ func Deploy(ctx context.Context, projects []project.Project, environmentClients
 	return nil
 }
 
+func Deploy(ctx context.Context, clientSet *client.ClientSet, projects []project.Project, sortedConfigs []graph.SortedComponent, environment string) error {
```

Comment:
As the `client.ClientSet` is created outside of this function (and even outside the loop in the calling function), is the memory used by caches actually freed after `Deploy(...)` is done? I guess we need to do this explicitly, either by ensuring that the client set goes out of scope and can be freed, or if this is not possible at the moment, by emptying the caches.

Reply:
Added a clear function 👍